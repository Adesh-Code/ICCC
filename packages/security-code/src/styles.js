const styles = ({ tokens, primitives }) => ({
  "&.DhigSecurityCode": {
    width: "56px",
    height: "72px",
    borderTop: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`,
    borderBottom: "none"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"][class*="MuiInputBase-inputAdornedEnd"]':
    {
      marginRight: "0"
    },
  "&.DhigSecurityCode[class*='MuiInput-root']:hover:before": {
    height: "2px",
    backgroundColor: tokens.colors["fill-engage"],
    boxShadow: "none",
    borderBottom: `${primitives.borderWidths.medium} solid ${primitives.colors.primary.black}`
  },
  "&.DhigSecurityCode:hover": {
    boxShadow: "none"
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]': {
    boxShadow: `0px 1px 0px 0px ${primitives.colors.primary.white}`
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]:before': {
    borderBottom: `1px solid ${primitives.colors.primary.black}`
  },
  "&.DhigSecurityCode input:focus": {
    boxShadow: `0px 2px 0px 0px ${tokens.colors["ink-focus"]} inset, 0px 0px 0px 2px ${tokens.colors["ink-focus"]} inset`
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]:not(.Mui-disabled)::before': {
    boxShadow: "none"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"]:before': {
    width: "56px"
  },
  "&.DhigSecurityCode input": {
    flexGrow: "1"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"]': {
    padding: "0",
    width: "56px",
    height: "72px",
    fontSize: `${primitives.fontSizeFallbacks["6"]}`,
    fontWeight: primitives.fontWeights.default
  },
  '&.DhigSecurityCode--dense [class*="MuiInputBase-input"]': {
    padding: "0",
    width: "32px",
    height: "40px",
    fontSize: `${primitives.fontSizeFallbacks["3"]}`,
    fontWeight: primitives.fontWeights.default
  },
  "&.DhigSecurityCode--dense": {
    width: "32px",
    height: "40px"
  },
  '&.DhigSecurityCode--error[class*="MuiInputBase-root"]:before': {
    borderBottom: `1px solid ${primitives.colors.accent.clay}`
  },
  "&.DhigSecurityCode--error, &.DhigSecurityCode:first-of-type.DhigSecurityCode--error, &.DhigSecurityCode:last-child.DhigSecurityCode--error, &.DhigSecurityCode--error:before":
    {
      borderColor: primitives.colors.accent.clay
    },
  '&.DhigSecurityCode--error[class*="MuiInputBase-root"]:after': {
    boxShadow: `0px 1px 0px 0px ${primitives.colors.accent.clay}`
  },
  "&.DhigSecurityCode:first-of-type": {
    borderLeft: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`
  },
  "&.DhigSecurityCode:last-child": {
    borderRight: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`
  },
  "&.DhigSecurityCode:last-child [class*='MuiDivider-root'], &.DhigSecurityCode:focus-within [class*='MuiDivider-root']":
    {
      display: "none"
    }
});

export default styles;
