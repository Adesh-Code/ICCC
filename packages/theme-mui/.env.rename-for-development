# To load the MUI X Pro license for local development, rename this file to .env and replace with the actual license string
DHIG_MUIXPRO_ENTRY=PASTE_MUI_X_PRO_LICENSE_STRING_HERE