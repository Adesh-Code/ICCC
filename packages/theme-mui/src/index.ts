// Themes
export { default as themeDark } from "./themes/dark";
export { default as themeLight } from "./themes/light";

// Utilities
export { configureClassnameGenerator } from "./classname-generator";
export { setLicenseKey } from "./set-license-key";
