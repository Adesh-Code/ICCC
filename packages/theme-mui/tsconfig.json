{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "outDir": "lib/esm",
    "resolveJsonModule": true,
    "declarationDir": "lib/types"
  },
  "include": ["src"]
}
