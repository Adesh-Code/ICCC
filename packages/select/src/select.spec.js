import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import MuiMenuItem from "@mui/material/MenuItem";
import Select from "./select";

import { themeLight } from "@iccc/theme-mui";
import { createTheme, ThemeProvider } from "@mui/material/styles";

const light = createTheme(themeLight);

const arrange = (otherProps) => {
  const args = {
    defaultValue: "",
    options: [
      { children: "child", value: "20" },
      { children: "child2", value: "30" }
    ],
    ...otherProps
  };

  return render(
    <ThemeProvider theme={light}>
      <Select {...args} />
    </ThemeProvider>
  );
};

describe("Select component", () => {
  it("renders", () => {
    const { container } = arrange();
    expect(container).toMatchSnapshot();
  });

  it("does not render if no options are provided", () => {
    const { container } = arrange({ options: null });
    expect(container).toBeEmptyDOMElement();
  });

  it("renders a selected icon", () => {
    const { getAllByTitle } = arrange({
      open: true,
      value: "20",
      defaultValue: "20"
    });
    expect(getAllByTitle("Selected")[0]).toBeInTheDocument();
  });

  it("accepts a fixed width", () => {
    const { container } = arrange({ width: "200px" });
    expect(container.firstChild).toHaveStyle({ width: "200px" });
  });

  it("accepts children", () => {
    const { getAllByRole } = render(
      <ThemeProvider theme={light}>
        <Select defaultValue="">
          <MuiMenuItem value={40}>Forty</MuiMenuItem>
          <MuiMenuItem value={20}>Twenty</MuiMenuItem>
        </Select>
      </ThemeProvider>
    );
    const select = getAllByRole("button")[0];
    fireEvent.mouseDown(select);
    expect(screen.getByText(`Forty`)).toBeInTheDocument();
  });
});
