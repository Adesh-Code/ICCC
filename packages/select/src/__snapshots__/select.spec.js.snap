// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select component renders 1`] = `
<div>
  <div
    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary  emotion-0"
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input emotion-1"
      role="button"
      tabindex="0"
    >
      <span
        class="notranslate"
      >
        ​
      </span>
    </div>
    <input
      aria-hidden="true"
      aria-invalid="false"
      class="MuiSelect-nativeInput emotion-2"
      tabindex="-1"
      value=""
    />
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit MuiSelect-icon MuiSelect-iconOutlined emotion-3"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <svg
        fill="none"
        viewBox="0 0 12 16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          id="caret-down"
        >
          <path
            d="M11.001 5.5L6.00098 10.5L1.00098 5.5"
            id="stroke"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
          />
        </g>
      </svg>
    </svg>
    <fieldset
      aria-hidden="true"
      class="MuiOutlinedInput-notchedOutline emotion-4"
    >
      <legend
        class="emotion-5"
      >
        <span
          class="notranslate"
        >
          ​
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
